pkg/kubelet/pod/mirror_client.go

```diff
@@ -82,6 +85,16 @@ func (mc *basicMirrorClient) DeleteMirrorPod(podFullName string) error {
 		klog.Errorf("Failed to parse a pod full name %q", podFullName)
 		return err
 	}
+	if uid != nil {
+		if apiPod, err := mc.apiserverClient.CoreV1().Pods(namespace).Get(name, metav1.GetOptions{}); err == nil {
+			klog.Infof("DeleteMirrorPod %s: requested UID %s actual %s", podFullName, *uid, apiPod.ObjectMeta.UID)
+			if apiPod.ObjectMeta.UID != *uid {
+				return fmt.Errorf("UID of pod %q (%s) does not match requested UID (%s); not deleting mirror pod", podFullName, apiPod.ObjectMeta.UID, *uid)
```

Comment:
Along with error, you can also return whether the error was due to UID mismatch.
This way, we don't rely on string content for this condition.

Reply:
See above comment about string content (which I've moved, for now at least, into DeleteMirrorPod).

I'm returning two values from DeleteMirrorPod, a bool to indicate whether the pod was actually deleted (so the caller knows whether it should try to recreate it) and an error if something other than not found or UID mismatch (which is really the same thing) happened.

Reply:
Fixed.